chain/client/src/sync.rs

```diff
@@ -258,15 +253,12 @@ impl HeaderSync {
     }
 
     fn get_locator(&mut self, chain: &mut Chain) -> Result<Vec<CryptoHash>, near_chain::Error> {
-        let tip = chain.sync_head()?;
-        let heights = get_locator_heights(tip.height);
-
-        // Clear history_locator in any case of header chain rollback.
```

Comment:
Is this logic not needed?

Reply:
the logic will always be false if we remove sync head and replace it with header head. So I just removed it.